### PR TITLE
[CELEBORN-1190][FOLLOWUP] Apply error prone patch and suppress some problems

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
1. Fix IntLongMath, InconsistentCapitalization, UnnecessaryAssignment
2. disable StringSplitter, EmptyBlockTag, EqualsGetClass, MissingSummary, BadImport

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
GA

```bash
./build/make-distribution.sh --release
```

PR

total  202

```
  43 SynchronizeOnNonFinalField
  24 StaticAssignmentInConstructor
  21 JdkObsolete
  16 ThreadLocalUsage
  16 MutableConstantField
  16 MissingCasesInEnumSwitch
  11 UnusedMethod
  11 NonAtomicVolatileUpdate
   8 UnusedNestedClass
   8 NonOverridingEquals
   8 Finally
   4 MixedMutabilityReturnType
   4 DoubleBraceInitialization
   4 CatchAndPrintStackTrace
   4 CanonicalDuration
   2 ReferenceEquality
   1 ClassCanBeStatic
   1 ByteBufferBackingArray
```

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java
Comment (pan3793) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:65]: so this is a bug?

Comment (pan3793) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:65]: does it cause a real impact?

Comment (cxzl25) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:65]: We just use different variable names here to avoid this problem.

https://errorprone.info/bugpattern/InconsistentCapitalization

Comment (pan3793) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:65]: Oh, I see

**Review by pan3793 — APPROVED** [review]

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Comment by FMX** [comment]
I have a question that the maven build system will be dropped after v0.5.0 or later. Is this necessary? Will error prone be integrated into SBT?

**Comment by cxzl25** [comment]
> Will error prone be integrated into SBT

A JIRA ticket has been created to track this issue. [CELEBORN-1080](https://issues.apache.org/jira/browse/CELEBORN-1080)

Tried it locally but not successful yet.

**Comment by FMX** [comment]
IMO, it would be better to hold this PR until there is a solution to add the error-prone plugin into SBT.

**Comment by waitinfuture** [comment]
> IMO, it would be better to hold this PR until there is a solution to add the error-prone plugin into SBT.

Hi @FMX , I think we don't require this PR to also support SBT before merging. It's totally OK to do it in future.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.5.0)/branch-0.4(v0.4.1)

**Pull request closed without merging**
